src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,776 @@
+    pyverilate_get_liveness_threshold_cycles,
+)
+from finn.util.data_packing import (
+    npy_to_rtlsim_input,
+    pack_innermost_dim_as_hex_string,
+    rtlsim_output_to_npy,
+)
+
+try:
+    from pyverilator import PyVerilator
+except ModuleNotFoundError:
+    PyVerilator = None
+
+"""@package Thresholding_rtl
+- ONNX i/o tensor shape assumptions for Thresholding:
+- input 0 is the input tensor, shape (..., NumChannels)
+- input 1 is the threshold tensor, shape (NumChannels, n_thres)
+- output 0 is the output tensor, shape (..., NumChannels) - same as input
+- the '...' here can be any shape (representing groups of vectors)
+
+This module creates an RTL IP, HLS is not supported. See 'thresholding_batch'
+for a HLS equivalent.
+"""
+
+
+class Thresholding_rtl(HLSCustomOp):
+    """Class that corresponds to finn-rtllib 'thresholding' function."""
+
+    def __init__(self, onnx_node, **kwargs):
+        super().__init__(onnx_node, **kwargs)
+
+    def get_nodeattr_types(self):
+        my_attrs = {
+            # parallelization; channels thresholded per cycle
+            "PE": ("i", True, 0),
+            # number of channels (each may have different thresholds)
+            "NumChannels": ("i", True, 0),
+            # number of steps in thresholding function. Used only in decoupled mode
+            "numSteps": ("i", True, 1),
+            # FINN DataTypes for inputs, outputs
+            "inputDataType": ("s", True, ""),
+            "weightDataType": ("s", True, ""),
+            "outputDataType": ("s", True, ""),
+            # number of input vectors, examples:
+            # [1] is a single vector (like a FC layer with batch=1)
+            # [4] is four vectors (like a FC layer with batch=4)
+            # [1, 4, 4] is four * four vectors (like a conv layer with batch=1)
+            "numInputVectors": ("ints", False, [1]),
+            # name of the top module in verilog template. Used by PyVerilator
+            # and IPI generation
+            "gen_top_module": ("s", False, ""),
+            # bias to be applied to outputs of the node
+            "activation_bias": ("i", False, 0),
+            # whether weights (thresholds) will be
+            # writable through an AXI-lite interface during runtime
+            # 1 for enabled, 0 for disabled.
+            "runtime_writeable_weights": ("i", False, 0, {0, 1}),
+            # memory depth triggers for threshold storage
+            "depth_trigger_uram": ("i", False, 0),
+            "depth_trigger_bram": ("i", False, 0),
+            # enable uniform thres optimization
+            # doesn't actually do anything yet, only
+            # for resource estimations
+            "uniform_thres": ("i", False, 0, {0, 1}),
+            # enable deep pipelining for easier timing closure
+            # setting to 0 may save some FFs but otherwise leave on
+            "deep_pipeline": ("i", False, 1, {0, 1}),
+        }
+        my_attrs.update(super().get_nodeattr_types())
```

Comment:
If you replace this line, with: 
`my_attrs.update(Thresholding.get_nodeattr_types(self))` 
 `my_attrs.update(RTLBackend.get_nodeattr_types(self))`
 
 Then the node attributes from the abstraction layer and the RTLBackend get correctly inserted.